### PR TITLE
Add a lot more docs for firehoses

### PR DESCRIPTION
**Pull request by fjy**

### CODE REVIEW

File: docs/content/ingestion/firehose.md [hunk 1]
```diff
@@ -19,14 +19,53 @@ Available Firehoses
 
 There are several firehoses readily available in Druid, some are meant for examples, others can be used directly in a production environment.
 
-#### KafkaFirehose
+#### KafkaEightFirehose
 
-This firehose acts as a Kafka consumer and ingests data from Kafka.
+Please note that the "kafka-0.8" [module](../operations/including-extensions.html) is required for this firehose. This firehose acts as a Kafka 0.8.x consumer and ingests data from Kafka.
```
Comment (nishantmonu51) [inline docs/content/ingestion/firehose.md]: specify correct extension name "druid-kafka-eight" instead of "kafka-0.8"

**Comment by nishantmonu51** [comment]
+1

File: docs/content/ingestion/firehose.md [hunk 2]
```diff
@@ -153,7 +214,7 @@ A sample ingest firehose spec is shown below -
 
 |property|description|required?|
 |--------|-----------|---------|
-|type|ingestSegment. Type of firehose|yes|
+|type|This should be "ingestSegment".|yes|
```
Comment (xvrl) [inline docs/content/ingestion/firehose.md]: can we use backticks for those, i.e. `` `ingestSegment` ``?

Comment (fjy) [inline docs/content/ingestion/firehose.md]: we dont do this convention anywhere else in the docs

File: docs/content/ingestion/firehose.md [hunk 3]
```diff
@@ -54,54 +94,74 @@ Sample spec:
...
+|maxEventCount|max events to receive, -1 is infinite, 0 means nothing is delivered; use this to prevent infinite space consumption or to prevent getting throttled at an inconvenient time.|N/A|yes|
+|maxRunMinutes|maximum number of minutes to fetch Twitter events.  Use this to prevent getting throttled at an inconvenient time. If zero or less, no time limit for run.|N/A|yes|
 
-#### RabbitMqFirehose
+#### RabbitMQFirehose
 
 This firehose ingests events from a define rabbit-mq queue.
-<br>
+
 **Note:** Add **amqp-client-3.2.1.jar** to lib directory of druid to use this firehose.
```
Comment (drcrallen) [inline docs/content/ingestion/firehose.md]: Should this just be adding to classpath rather than to lib directory?

Comment (fjy) [inline docs/content/ingestion/firehose.md]: I'm not sure, I've never used the rabbitmq firehose, I'd rather just keep whta ppl have been successful with

**Pull request merged**